Managers/ConsoleMan.cpp

```diff
@@ -1,541 +1,423 @@
-    {
-        m_pParentBox->SetEnabled(true);
-        m_pParentBox->SetVisible(true);
-
-        float toGo = -floorf((float)m_pParentBox->GetYPos() * MENUSPEED);
-        m_pParentBox->SetPositionAbs(0, m_pParentBox->GetYPos() + toGo);
-
-        if (m_pParentBox->GetYPos() >= 0)
-            m_EnabledState = ENABLED;
-    }
-    // Animate the menu out of view
-    else if (m_EnabledState == DISABLING)
-    {
-        float toGo = -ceilf(((float)m_pParentBox->GetHeight() + (float)m_pParentBox->GetYPos()) * MENUSPEED);
-        m_pParentBox->SetPositionAbs(0, m_pParentBox->GetYPos() + toGo);
-
-        if (m_pParentBox->GetYPos() <= -m_pParentBox->GetHeight())
-        {
-            m_pParentBox->SetEnabled(false);
-            m_pParentBox->SetVisible(false);
-            m_EnabledState = DISABLED;
-        }
-    }
-    else if (m_EnabledState == ENABLED)
-    {
-        // If supposed to be enabled but appears to be disabled, enable it gracefully
-        if (m_pParentBox->GetYPos() < 0)
-            m_EnabledState = ENABLING;
-
-        m_pParentBox->SetEnabled(true);
-        m_pParentBox->SetVisible(true);
-        m_pInputTextBox->SetEnabled(true);
-        m_pInputTextBox->SetFocus();
-    }
-    else if (m_EnabledState == DISABLED)
-    {
-		if (!m_ForceVisible)
-		{
-			// If supposed to be disabled but appears to be out anyway, disable it gracefully
-			if (m_pParentBox->GetYPos() > -m_pParentBox->GetHeight())
-			{
-				m_pParentBox->SetEnabled(true);
-				m_pParentBox->SetVisible(true);
-				m_EnabledState = DISABLING;
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::SetEnabled(bool enable) {
+		if (enable && m_ConsoleState != ENABLED && m_ConsoleState != ENABLING) {
+			m_ConsoleState = ENABLING;
+			g_GUISound.EnterMenuSound()->Play();
+		} else if (!enable && m_ConsoleState != DISABLED && m_ConsoleState != DISABLING) {
+			m_ConsoleState = DISABLING;
+			g_GUISound.ExitMenuSound()->Play();
+		}
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+	
+	void ConsoleMan::SetReadOnly() {
+		if (m_ReadOnly) {
+			return;
+		}
+		// Save the current input string before changing to the read-only note so we can restore it when switching back
+		m_LastInputString = m_InputTextBox->GetText();
+		m_InputTextBox->SetText("- CONSOLE IN READ-ONLY MODE! -");
+		m_ReadOnly = true;
+		SetEnabled(true);
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ConsoleMan::SetConsoleScreenSize(float screenRatio) {
+		m_ConsoleScreenRatio = Limit(screenRatio, 1.0F, 0.1F);
+
+		if (!m_ParentBox || !m_ConsoleText || !m_InputTextBox) {
+			return;
+		}
+
+		// Stretch the parent box to fill the top of the screen widthwise, and as far down as the parameter says
```

Comment:
Don't think we need any of these comments. The first one is straightforward and the 2nd doesn't actually make it any easier to understand imo - in that it's vague and doesn't explain things. If it (the 2nd comment) explained more specifically what the code was doing then that'd mean there's no need to read the code I guess, but it doesn't so it's kinda pointless.